### ISSUE TITLE
Can I write an npm package compatible with preact and react?

### ISSUE THREAD

**Issue by qwertie** [developit/preact-compat#497]
As I was typing `npm install react`, I thought to myself: I'm sure my package would be preact-compatible, but if anybody writes `npm i mypackage` they are going to get `react` as a dependency, which maybe they don't want. Is there a way I can publish this thing in such a way that it's compatible with both?

**Comment by marvinhagemeister** [comment]
Third party packages usually opt for one of the following ways to be compatible with react/preact:

1. Kepp importing react and tell users to overwrite react import via webpack aliasing
2. Publish dedicated preact entrypoint `mypackage/preact`
3. Publish a completely separate package `mypackage-preact` + `mypackage-react`

**Comment by marvinhagemeister** [comment]
For all options you'd mark `react` or `preact` as a `peerDependency`. Otherwise the user would likely end up with multiple versions of (p)react

**Comment by developit** [comment]
You might find the technique shown here to be useful:
https://jasonformat.com/universal-vdom-components-with-factory-loader/

**Comment by qwertie** [comment]
I don't think that technique would work in TypeScript (which I'm using) as the compiler would be unable to load the pseudo-module. And I doubt the technique could be used with [Parcel](http://parceljs.org/) (which I'm also using). Hmm...
